ODHack: Fix Node Workflow Build Issue

### **Please add PRs to the [update-P21 branch](https://github.com/stellar/soroban-example-dapp/tree/update-P21)**

## Title: 

**Description:**
Resolve the build issue in the Node workflow where the `next build` command fails due to a type error in `contracts.ts` during a workflow check in the following pull request: https://github.com/stellar/soroban-example-dapp/pull/154

**Error Log:**
```
Failed to compile.

./shared/contracts.ts:8:25
Type error: Property 'testnet' does not exist on type '{ readonly futurenet: { readonly networkPassphrase: "Test SDF Future Network ; October 2022"; readonly contractId: "CAU23MVDEW7VMB3AXC5PL5GZO3HBYZPEIP2P46X4T3AIC4Y7FXYZWES5"; }; }'.

6 |
7 | export const abundance = new Abundance.Client({

8 | ...Abundance.networks.testnet,
| ^
9 | rpcUrl: rpcUrl,
10 | allowHttp: true,
11 |
Error: Process completed with exit code 1.
```

**Tasks:**
1. **Review Current Implementation:**
   - Review the `shared/contracts.ts` file to identify the source of the type error.
   - Verify the available networks in the `Abundance` client.

2. **Update Network Configuration:**
   - Correct the network configuration to use the appropriate network available in the `Abundance` client.
   - Ensure the `testnet` network configuration is correctly defined or replace it with an available network.

3. **Configure Build Caching:**
   - Configure build caching for Next.js to improve build times and provide faster rebuilds.
   - Follow the guidelines in the [Next.js documentation](https://nextjs.org/docs/messages/no-cache) for setting up build caching.

4. **Test the Build:**
   - Run `npm run build` to ensure the build completes successfully without errors.
   - Verify that the application runs correctly after the build.
   - Create a PR to test the workflow on Github

**Expected Outcome:**
- The type error in `shared/contracts.ts` is resolved.
- The build completes successfully without errors.
- The application runs correctly post-build.
- Improved build times with build caching configured.

**Why This Is Important:**
Fixing this build issue is crucial to ensure the continuous integration and deployment pipeline functions correctly, improving development efficiency and reliability.

--- a/contracts/abundance/src/test.rs
+++ b/contracts/abundance/src/test.rs
@@ -250,4 +250,3 @@ fn test_zero_allowance() {
     token.transfer_from(&spender, &from, &spender, &0);
     assert!(token.get_allowance(&from, &spender).is_none());
 }
-

--- a/contracts/abundance/src/test.rs
+++ b/contracts/abundance/src/test.rs
@@ -251,17 +251,3 @@ fn test_zero_allowance() {
     assert!(token.get_allowance(&from, &spender).is_none());
 }
 
-#[test]
-fn test_zero_allowance() {
-    // Here we test that transfer_from with a 0 amount does not create an empty allowance
-    let e = Env::default();
-    e.mock_all_auths();
-
-    let admin = Address::generate(&e);
-    let spender = Address::generate(&e);
-    let from = Address::generate(&e);
-    let token = create_token(&e, &admin);
-
-    token.transfer_from(&spender, &from, &spender, &0);
-    assert!(token.get_allowance(&from, &spender).is_none());
-}

